src/api/ddoc/views.rst

```diff
@@ -315,6 +315,1144 @@ including the update sequence of the database from which the view was
+
+*Using a guard clause to check whether the required data field exists,
+and holds a number,
+before an attempt to index:*
+
+.. code-block:: javascript
+
+    if (typeof(doc.min_length) === 'number') {
+	    index("min_length", doc.min_length, {"store": true});
+    }
+
+Use a generic guard clause test to ensure that the type of the candidate data field is defined.
+
+*Example of a 'generic' guard clause:*
+
+.. code-block:: javascript
+
+    if (typeof(doc.min_length) !== 'undefined') {
+	    // The field exists, and does have a type, so we can proceed to index using it.
+	    ...
+    }
+
+.. _api/ddoc/view/analyzers:
+
+Analyzers
+---------
+
+Analyzers are settings that define how to recognize terms within text.
+Analyzers can be helpful if you need to :ref:`language-specific-analyzers <api/ddoc/view>`.
+
+Here's the list of generic analyzers that are supported by search:
+
++----------------+---------------------------------------------------------------------------------+
+| Analyzer       | Description                                                                     |
++================+=================================================================================+
+| ``classic``    | The standard Lucene analyzer, circa release 3.1.                                |
++----------------+---------------------------------------------------------------------------------+
+| ``email``      | Like the ``standard`` analyzer, but tries harder to match an email              |
+|                | address as a complete token.                                                    |
++----------------+---------------------------------------------------------------------------------+
+| ``keyword``    | Input is not tokenized at all.                                                  |
++----------------+---------------------------------------------------------------------------------+
+| ``simple``     | Divides text at non-letters.                                                    |
++----------------+---------------------------------------------------------------------------------+
+| ``standard``   | The default analyzer. It implements the Word Break rules from the               |
+|                | `Unicode Text Segmentation algorithm <http://www.unicode.org/reports/tr29/>`_.  |
++----------------+---------------------------------------------------------------------------------+
+| ``whitespace`` | Divides text at white space boundaries.                                         |
++----------------+---------------------------------------------------------------------------------+
+
+
+*Example analyzer document:*
+
+.. code-block:: javascript
+
+    {
+	    "_id": "_design/analyzer_example",
+	    "indexes": {
+		    "INDEX_NAME": {
+			    "index": "function (doc) { ... }",
+			    "analyzer": "$ANALYZER_NAME"
+		    }
+	    }
+    }
+
+.. _api/ddoc/view/language-specific-analyzers:
+
+Language-specific analyzers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
These all come from (a specific version of) Lucene, yes? Perhaps we should include a link back to Lucene's documentation where the full list is also specified.

Reply:
Still needs resolution.

Reply:
For clarity, this still needs a link back to Apache Lucene's documentation somewhere.

Reply:
It's not obvious to me what the best link would be. We could use the javadoc for `org.apache.lucene.analysis.Analyzer` from which all of these inherit:

https://lucene.apache.org/core/4_6_1/core/org/apache/lucene/analysis/Analyzer.html

We could use the package documentation for Analysis, which goes into a bunch of detail about how to use these analyzers and what they do -- most of which is not at all relevant for CouchDB users:

https://lucene.apache.org/core/4_6_1/core/org/apache/lucene/analysis/package-summary.html

I guess it doesn't really matter much. I just don't want to send people down a rabbit hole, so let's make sure we're linking to reference material as opposed to required reading.